pandas/core/arrays/boolean.py

```diff
@@ -688,6 +688,35 @@ def cmp_method(self, other):
         name = f"__{op.__name__}"
         return set_function_name(cmp_method, name, cls)
 
+    def _accumulate(
```

Comment:
I suspect this can be put on `pandas.core.arrays.masked.BaseMaskedArray`.

Reply:
May also want to look at `core.array_algos.masked_reductions` for inspiration. I suspect it'll look somewhat similar.

Reply:
This is a good idea to deduplicate the code for boolean and integer arrays. Thanks for the hint

Reply:
Actually, for deduplication, how about defining this as

```python
from pandas.arrays import IntegerArray
return IntegerArray(self._data.astype(int), self._mask)._accumulate(name, skipna, **kwargs)
```

Reply:
I guess this would upcast a boolean array always to an integer array. This might not be desired for `cummax`, `cummin` and `cumprod`. However, I'd be fine with that approach

Reply:
I chose 'int8' as dtype to keep the arrays as small as possible